### DISCUSSION TITLE
trino read file from s3a

### DISCUSSION THREAD

**Discussion by JunseoChoJJ** [trinodb/trino#21717] [Q&A]
Hello,

I really need your help. I am newbee to iceberg and trino

My architecture is trino iceberg s3a on kubernetes.
I did iceberg connector and it works.
i can create iceberg table and show it in my object storage
however, if i want to read my csv data from object storage, it gets error. I know that https://trino.io/docs/current/connector/iceberg.html trino iceberg connector allows 3 types of file format 
so my question is the easiest way is put parquet file into my object storage and read 
but I want to know is there any way to read csv data file from obs using iceberg connector( I think question is awkward)
maybe put hive metastore connector that i can read csv data from object storage
 does anyone put two connectors? (It seems not inefficent I guess) 

create table hive.trino_test.csv_table (
	no int,
	user varchar,
	salary int
)
with (format = 'csv', location = 's3a://trino_test/data/a.csv')

**Comment by tlm365** [comment]
How does that CSV file exist in S3? I mean what method do you use to ingest that CSV file?
I suggest use [Glue Catalog as metastore](https://trino.io/docs/current/object-storage/metastores.html#aws-glue-catalog-configuration-properties) for data in S3 (for both Iceberg and other formats) and [Hive as connector](https://trino.io/docs/current/connector/hive.html#general-configuration).
For Iceberg, you need change catalog to [Glue Catalog](https://iceberg.apache.org/docs/1.5.0/aws/#glue-catalog)
For CSV and other file formats you can use Glue crawler or write it as Athena/Glue table by [awswrangler](https://aws-sdk-pandas.readthedocs.io/en/3.2.1/stubs/awswrangler.s3.to_csv.html) or [Glue spark](https://docs.aws.amazon.com/glue/latest/dg/aws-glue-programming-etl-format-csv-home.html), then you can read it easily

**Reply by JunseoChoJJ** [reply]
@tlm365  thanks for the reply first
I know that s3 can collect any file format which is datalake? (my concept could be wrong) I just upload csv files  in my object storage in web. and I want to read data from it using trino and I want to tag that data into iceberg and create iceberg table
 
how about how can I read parquet file in s3a. I need to use hive catalog since we're not using aws.  
is there no way to read data from obs and create an iceberg table?
[https://github.com/trinodb/trino/discussions/20329 I want to do something like this trino obs iceberg](https://github.com/trinodb/trino/discussions/19042)

here is the example that I want to do
